web-ng/src/app/services/layer/layer.service.ts

```diff
@@ -83,7 +83,11 @@ export class LayerService {
   /**
    * Adds/Updates the layer of a project with a given layer value.
    */
-  addOrUpdateLayer(projectId: string, layer: Layer): Promise<void> {
+  async addOrUpdateLayer(projectId: string, layer: Layer): Promise<void> {
+    if (layer.index === -1) {
+      const layerCount = await this.getLayerCount();
+      (layer.index as number) = layerCount;
```

Comment:
I know you might ask why I am casting here, the reason is index is readonly property and I cannot assign it directly. Another approach could be creating new Layer instance which I don't prefer.

Reply:
Unfortunately this breaks immutability of the object; we rely on the assumption that our model objects are immutable, and this can lead to undesired consequences elsewhere. Instead, I would define a new method in Layer, `withIndex(number)`, that would return a copy of the Layer with the new index set. Thanks!!

Reply:
appreciate it 😁